### ISSUE TITLE
Refactor gradle for gitFs

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#3398]
I think we should soon work under the assumption that running Gradle requires gitFs. i.e. all files will be available in the repository, without special action being taken to include them.

This may mean we reconsider how we define `fileMatch` and therefore `packageFiles`. i.e. this list should really only be the files that define dependencies and not necessarily every other file we need "just in case".

Does this also change our ability to support a `build.gradle` outside the root?

Raising this to discuss with @corecanarias in advance of migrating to gitFs-only soon.

Waiting on #2812

**Comment by corecanarias** [comment]
I'm not sure if it makes sense to invest efforts on this, at least until we find out how parsing build.gradle ourselves work. That will remove the necessity of running gradle to retrieve the dependencies. I guess this will put gradle support at the same level as other package managers.

If there are too many cases that we cannot support, probably it would make sense to support both ways. But I would try to keep it simple and only maintain one.

**Comment by rarkins** [comment]
I think this is already partly done, and also it probably doesn't need its own separate issue for tracking.
